### ISSUE TITLE
XAML Designer creates incorrect PCL errors

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#1373]
Open https://github.com/SLaks/DroidMaster/blob/master/DroidMaster/Core/PersistentDevice.cs, then open https://github.com/SLaks/DroidMaster/blob/master/DroidMaster/UI/SourcePicker.xaml in the XAML designer, and start editing the XAML code.

After a few seconds, errors like this will appear

 > The type 'IDisposable' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.	Core\PersistentDevice.cs	

The project still builds fine.
The errors do not go away until their source file is closed (even after closing the XAML designer).

**Comment by Pilchie** [comment]
Can you take a look and see if this is related to the targets file changes around Facade assemblies, and if it should be transitioned to the designer team?

**Comment by Pilchie** [comment]
Okay - we believe we know what is causing this issue.  You should be able to work around it by adding an MSBuild property to your project:
``` XML
<CheckForSystemRuntimeDependency>true</CheckForSystemRuntimeDependency>
```

We'll keep looking to see what the right fix is on our end.

**Comment by SLaks** [comment]
Worked!
Thanks

**Comment by Pilchie** [comment]
Sadly this fix was considered too risky for the the rc2 branch, so it will remain 1.0 only.

**Comment by Pilchie** [comment]
Rather than leaving this open, now that I've confirmed that the same thing fixes it, I'm just going to close this as a duplicate of #560.

**Comment by Pilchie** [comment]
Can you verify this?  It's related to the internal bug I just gave you to verify too.

**Comment by dpoeschl** [comment]
Verified against Monday's ```1.0 (stable)``` build.

**Comment by mcosmin222** [comment]
This problem occurs with almost 100% chance when dealing with break points in async void methods. Complains about missing reference

error CS0012: The type 'TaskAwaiter<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Threading.Tasks, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
